File: spring-boot-java-web/src/main/java/com/example/webjsp/controller/HomeController.java
package com.example.webjsp.controller;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.i18n.LocaleContextHolder;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.CookieValue;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.servlet.LocaleContextResolver;
import org.springframework.web.servlet.LocaleResolver;
import org.springframework.web.servlet.i18n.CookieLocaleResolver;

@Slf4j
@Controller
public class HomeController {

    @GetMapping("/")
    public String index(Model model) {
        String userLocale = LocaleContextHolder.getLocale().getLanguage();
        log.info("/index: {}\n{}", userLocale, LocaleContextHolder.getLocale());
        model.addAttribute("message", "Welcome to Simple Solution home page!");
        return "index";
    }

    @GetMapping("/user/list")
    public String userList(Model model) {
        String userLocale = LocaleContextHolder.getLocale().getLanguage();
        log.info("/user/list: {}", userLocale);
        if (model == null) {
            log.warn("model not be null");
        }
        return "user/list";
    }
}

File: spring-boot-kotlin-web/src/main/kotlin/com/example/webjsp/controller/HomeController.kt
package com.example.webjsp.controller

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.context.i18n.LocaleContextHolder
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.CookieValue
import org.springframework.web.bind.annotation.GetMapping

@Controller
class HomeController {
  @GetMapping("/")
  fun index(model: Model): String {
    val userLocale = LocaleContextHolder.getLocale().language;
    log.info("current lang: {}", userLocale)
    model.addAttribute("message", "Welcome to Simple Solution home page!")
    return "index"
  }

  @GetMapping("/user/list")
  fun userList(model: Model): String {
    val userLocale = LocaleContextHolder.getLocale().language;
    log.info("current lang: {}", userLocale)
    if (model == null) {
      log.warn("model not be null")
    }
    return "user/list"
  }

  companion object {
    val log: Logger = LoggerFactory.getLogger(HomeController::class.java)
  }
}
